lms/djangoapps/bulk_email/views.py

```diff
@@ -25,47 +25,46 @@
 log = logging.getLogger(__name__)
 
 
-def opt_out_email_updates(request, token, course_id):
+def opt_out_email_updates(request, token):
     """
     A view that let users opt out of any email updates.
 
     This meant is meant to be the target of an opt-out link or button.
-    The `token` parameter must decrypt to a valid username.
-    The `course_id` is the string course key of any course.
+    The `token` parameter must decrypt to a valid username and course id.
 
     Raises a 404 if there are any errors parsing the input.
     """
+
     try:
-        username = UsernameCipher().decrypt(token.encode())
-        user = User.objects.get(username=username)
+        email, course_id = UsernameCipher().decrypt(token).decode().split(' ')
```

Comment:
AT some places we are encrypting a username along with course id instead of an email but I don't see any code handling that case.

Also is there any specific reason we are changing existing logic to build unsubscribe links?